### PR TITLE
Adding js/lib/*.js and removing from gitignore

### PR DESCRIPTION
**Pull request by MatthewTurk**
I've (manually) added the cookiecutter stuff for the lib subdirectory and updated gitignore to ... not ignore them.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -121,20 +121,20 @@ def run(self):
...
     'version': version_ns['__version__'],
     'description': 'A Custom Jupyter Widget Library',
     'long_description': LONG_DESCRIPTION,
     'include_package_data': True,
     'data_files': [
         ('share/jupyter/nbextensions/yt-jscanvas', [
-            'yt-pycanvas/static/extension.js',
-            'yt-pycanvas/static/index.js',
-            'yt-pycanvas/static/index.js.map',
+            'yt_pycanvas/static/extension.js',
```
Comment (munkm) [inline setup.py]: I noticed that you changed the name to yt_pycanvas from yt-pycanvas. The line above this one has the javascript package that we named for the canvas widget `yt-jscanvas`. Should we be consistent with the underscoring vs. hyphen, or is there a best practice with using one over the other?

Comment (MatthewTurk) [reply setup.py]: Ah!  Good question.  I think it's a big deal with python, but not with javascript, maybe?  I was unable to get the library to import unless I changed it.

Comment (munkm) [reply setup.py]: Weird! Well, maybe down the line I can update the javascript lib so they have consistent naming conventions.

**Comment by munkm** [comment]
Ok I've successfully used and installed this version of the widget with HelloWorld. 👍

**Pull request merged**